client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx

```diff
@@ -143,56 +143,72 @@ const UnifiedDesignPickerStep: Step = ( { navigation, flow } ) => {
 	const selectedDesign = useSelect( ( select ) => select( ONBOARD_STORE ).getSelectedDesign() );
 	const { setSelectedDesign } = useDispatch( ONBOARD_STORE );
 
-	function getEventPropsByDesign( design: Design ) {
+	const isEnabledStyleSelection = isEnabled( 'signup/design-picker-style-selection' );
+	const selectedDesignHasStyleVariations =
+		isEnabledStyleSelection &&
+		selectedDesign?.style_variations &&
+		selectedDesign?.style_variations.length > 0;
+
+	const { data: selectedDesignDetails } = useStarterDesignBySlug( selectedDesign?.slug || '', {
+		enabled: isPreviewingDesign && selectedDesignHasStyleVariations,
+	} );
+
+	const selectedStyleVariation = useSelect( ( select ) =>
+		select( ONBOARD_STORE ).getSelectedStyleVariation()
+	);
+	const { setSelectedStyleVariation } = useDispatch( ONBOARD_STORE );
+
+	function getEventPropsByDesign( design: Design, variation?: StyleVariation ) {
+		const variationSlugSuffix =
+			variation && variation.slug !== 'default' ? `-${ variation.slug }` : '';
+
 		return {
-			slug: design?.slug,
-			theme: design?.recipe?.stylesheet,
 			flow,
 			intent,
-			is_premium: design?.is_premium,
+			slug: design.slug + variationSlugSuffix,
+			theme: design.recipe?.stylesheet,
+			theme_style: design.recipe?.stylesheet + variationSlugSuffix,
 			design_type: design.design_type,
-			...( design?.recipe?.pattern_ids && { pattern_ids: design.recipe.pattern_ids.join( ',' ) } ),
-			...( design?.recipe?.header_pattern_ids && {
+			is_premium: design.is_premium,
+			...( design.recipe?.pattern_ids && { pattern_ids: design.recipe.pattern_ids.join( ',' ) } ),
+			...( design.recipe?.header_pattern_ids && {
 				header_pattern_ids: design.recipe.header_pattern_ids.join( ',' ),
 			} ),
-			...( design?.recipe?.footer_pattern_ids && {
+			...( design.recipe?.footer_pattern_ids && {
 				footer_pattern_ids: design.recipe.footer_pattern_ids.join( ',' ),
 			} ),
+			has_style_variations:
```

Comment:
Small detail, but for generated design preview `has_style_variations` will be `undefined` and thus the event won't have this prop. Not sure if it's a concern at the moment, that it'd be good to add it as `false` for consistency.

Reply:
Nice find again 🤦 JS quirks. Fixed!

Reply:
Agreed. :)